experimental/dds/tree/src/EditLog.ts

```diff
@@ -476,14 +563,18 @@ export class EditLog<TChange = unknown> extends TypedEventEmitter<IEditLogEvents
 	 * Assigns provided handles to edit chunks based on chunk index specified.
 	 */
 	public processEditChunkHandle(chunkHandle: EditHandle<TChange>, startRevision: number): void {
-		const chunk = this.editChunks.get(startRevision);
+		assert(startRevision >= this.firstAvailableIndex, 'the revision has already been evicted');
+
+        const chunk = this.editChunks.get(startRevision);
 		if (chunk !== undefined) {
 			assertNotUndefined(
 				chunk.edits,
 				'A chunk handle op should not be received before the edit ops it corresponds to.'
 			);
 			chunk.handle = chunkHandle;
-			this.addKeyToCache(startRevision);
+            if (startRevision < this.indexOfFirstEditInSession) {
```

Comment:
Same point here. I think we should probably just throw an exception if this is true ("log eviction and chunking are mutually exclusive" or something like that). Otherwise, we will never drop the edits field.